### ISSUE TITLE
Remove concrete application examples from API

### ISSUE THREAD

**Issue by tobiastom** [grand-decentral-station/concept#30]
Right now inside the API.md there is a concrete example about building a distributed social network with GDS.

**IMHO** this is way to specific. For me GSD is a concept to install and delete applications and have them communicate with each other. GDS should **only** define the API between applications on **one** GDS node. 

Connections to other GDS nodes should **never** be defined by this protocol. 

To get back to the distributed social network example in API.md, there could be a GDS application that implements the [tent](http://tent.io) protocol to archive that. One application at a time.

**Comment by lukasbestle** [comment]
APIs between multiple GDS nodes are important, because there are multiple ways to implement each task on the server (let's take a shared calendar: if every app defines its own protocols, this will most likely fail).

**Comment by tobiastom** [comment]
I completely agree, but we really should not throw away all the great work that was done with CalDAV, CardDAV and e.g. XMPP.

Again: **I** think we should define how the applications run on the server, now what they are running.

**Comment by lukasbestle** [comment]
Of course, we should use as much existing protocols as possible.
Still, there are basic protocols we need to define and make available to all application such as discoverability.

**Comment by tobiastom** [comment]
Like intends: definitely yes. What else do you think we need?

Am 30.10.2013 um 15:00 schrieb Lukas Bestle notifications@github.com:

**Comment by janl** [comment]
On that note, http://sockethub.org seems to be a nice bridge to protocols

**Comment by augustl** [comment]
I agree. GDS isn't ownCloud. :) The default GDS implementation we create can come bundled with some apps, batteries included, Android style. But as much as possible should be replaceable. Perhaps we can go as far as having the UI for managing permissions use protocols and something intent-like (so the system knows which app that implements "calendar" and can prompt the user for which app to open if there are multiple "calendar" intent apps).

**Comment by tobiastom** [comment]
Exactly. I could not agree more

**Comment by augustl** [comment]
Just to throw it out there, perhaps GDS apps should be able to communicate and use intents for non-GDS apps. Then the only thing GDS helps with is a dashboard for your apps, installation, etc.

https://en.wikipedia.org/wiki/Web_Intents

**Comment by tobiastom** [comment]
It might even be an option to have even all GDS applications on one server only communicate via intends (or WebActivities).

**Comment by bastianallgeier** [comment]
Just a question. Wouldn't it make sense to start writing down ideas for protocols or APIs. Pretty much like the hoodie guys do: http://nobackend.org/dreamcode.html

I think it would be much easier to discuss that.

**Comment by augustl** [comment]
@bastianallgeier I agree. Seems the consensus is a protocol driven approach. Perhaps a wiki is good for this, where we can throw around suggestions of various sets of protocols, where the set contains existing protocols and any new protocols that might need to be invented.
